gallery.js

```diff
@@ -73,10 +73,23 @@ document.addEventListener('DOMContentLoaded', () => {
         alert('Failed to copy!');
       }
     };
-    buttonsDiv.appendChild(copyBtn);
-
-
+    buttonsDiv.appendChild(copyBtn)
+    
+    //Delete Button
+    const deleteBtn = document.createElement('button');
+    deleteBtn.textContent = "Delete";
+    deleteBtn.addEventListener("click", function() {
+      if (confirm("Are you sure you want to delete this image? This action cannot be undone.")) {
+        card.remove();
+        localStorage.removeItem(imageMetadata.imageId); //removes image by imageId
+        let imageList = JSON.parse(localStorage.getItem('images') || '[]');
+        imageList = imageList.filter(image => image.imageId !== imageMetadata.imageId); //filtering out the removed imageId
+        localStorage.setItem('images', JSON.stringify(imageList)); //updating localStorage
```

Comment:
The image list is contained at the "`album_art_app_image_list"`, not the "images" key, so this doesn't retrieve and update the image list properly. The variable `imageListKey` contains the local storage key for it

```suggestion
        let imageList = JSON.parse(localStorage.getItem(imageListKey) || '[]');
        imageList = imageList.filter(image => image.imageId !== imageMetadata.imageId); //filtering out the removed imageId
        localStorage.setItem(imageListKey, JSON.stringify(imageList)); //updating localStorage
```